llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -788,13 +793,13 @@ void VPlan::prepareToExecute(Value *TripCountV, Value *VectorTripCountV,
   }
 
   for (unsigned Part = 0, UF = State.UF; Part < UF; ++Part)
-    State.set(&VectorTripCount, VectorTripCountV, Part);
+    State.set(&VectorTripCount, VectorTripCountV, VPIteration(Part, 0));
```

Comment:
Independent of this patch: would be good to be consistent if/when a scalar that is  invariant or uniform-across-parts is set in lane zero of all parts (e.g., VectorTripCount) or in lane zero of part zero only (e.g., VFxUF).